### PR TITLE
Fix PIN size according to the right maximum value

### LINKED ISSUE #7: PIN size can't be higher than a int size
Hi !

As long as, in Java, a int has a limited size, the function "intToByteArray" can't handle pin code higher than 2147483647.

I'm a newbie in Java, but is there a way to easily fix it ?

Thank you !

### PR DESCRIPTION
**Pull request by lonk**
Fixes #7

### CODE REVIEW

File: src/android/MifareUltralight.java [hunk 1]
```diff
@@ -22,11 +22,11 @@ public void disconnect() throws Exception {
...
     }
 
-    public boolean unlockWithPin(int pin) throws Exception {
-        final byte[] pinAsByteArray = intToByteArray(pin);
+    public boolean unlockWithPin(long pin) throws Exception {
+        final byte[] pinAsByteArray = longToByteArray(pin);
         byte[] response = mifare.transceive(new byte[]{
                 (byte) 0x1B, // PWD_AUTH
-                pinAsByteArray[0], pinAsByteArray[1], pinAsByteArray[2], pinAsByteArray[3]
+                pinAsByteArray[4], pinAsByteArray[5], pinAsByteArray[6], pinAsByteArray[7]
```
Comment (RoopeHakulinen) [inline src/android/MifareUltralight.java:29]: Probably I'm just missing something here but why we are sending only half of the pin here? 🤔

**Review by RoopeHakulinen — CHANGES_REQUESTED** [review]

File: src/android/MifareUltralight.java (hunk 1 above)
Comment (lonk) [reply src/android/MifareUltralight.java:29]: The max size of the PIN is 0xFFFFFFFF, so the 4 first bytes would be ignored by the chip !

Comment (RoopeHakulinen) [reply src/android/MifareUltralight.java:29]: Hmm.. I think I get what you mean so that should be fine.

**Comment by RoopeHakulinen** [comment]
Just to confirm one final thing @lonk: this is not a breaking change, right? So the existing usage of unlocking will work okay the way I see it? The usage is from JS only and there you only have the `number` type anyways.

**Comment by lonk** [comment]
It doesn't change the way unlocking was previously working !
The only change is that it doesn't throw an error anymore if the PIN is higher than 0x7FFFFFFF, and it truncates the PIN if it's higher than 0xFFFFFFFF (as an example: 0xAABBCCDDEE will try to unlock the tag with the pin 0xBBCCDDEE)

**Comment by RoopeHakulinen** [comment]
Excellent. I'll try to publish a new version soon.

**Pull request merged**